app/controllers/course/users_controller.rb

```diff
@@ -4,12 +4,24 @@ class Course::UsersController < Course::ComponentController
   include Course::UsersControllerManagementConcern
 
   before_action :load_resource
+  before_action :load_skills, only: [:show]
```

Comment:
There is quite some logic in load skills, suggest we move that into a service?

Reply:
Suggest you take a step back and think about the input parameters of the service and what should be the output/APIs.

Reply:
Moved to a service.